src/pgtest/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+                                    Ok(Field {
+                                        name: f.name().to_string(),
+                                    })
+                                })
+                                .collect()
+                                .unwrap(),
+                        })?,
+                    ),
+                    Message::DataRow(body) => {
+                        let buf = body.buffer();
+                        (
+                            "DataRow",
+                            serde_json::to_string(&DataRow {
+                                fields: body
+                                    .ranges()
+                                    .map(|range| {
+                                        let range = range.unwrap();
+                                        // TODO(mjibson): support not strings.
+                                        Ok(String::from_utf8(buf[range.start..range.end].to_vec())
+                                            .unwrap())
+                                    })
+                                    .collect()
+                                    .unwrap(),
+                            })?,
+                        )
+                    }
+                    Message::CommandComplete(body) => (
+                        "CommandComplete",
+                        serde_json::to_string(&CommandComplete {
+                            tag: body.tag().unwrap().to_string(),
+                        })?,
+                    ),
+                    Message::ParseComplete => ("ParseComplete", "".to_string()),
+                    Message::BindComplete => ("BindComplete", "".to_string()),
+                    Message::PortalSuspended => ("PortalSuspended", "".to_string()),
+                    Message::ErrorResponse(body) => (
+                        "ErrorResponse",
+                        serde_json::to_string(&ErrorResponse {
+                            fields: body
+                                .fields()
+                                .map(|f| {
+                                    Ok(ErrorField {
+                                        typ: f.type_(),
+                                        value: f.value().to_string(),
+                                    })
+                                })
+                                .collect()
+                                .unwrap(),
+                        })?,
+                    ),
+                    _ => ("UNKNOWN", format!("'{}'", ch)),
+                };
+                let mut s = typ.to_string();
+                if !args.is_empty() {
+                    s.push(' ');
+                    s.push_str(&args);
+                }
+                msgs.push(s);
+                if expect == typ {
+                    break;
+                }
+            }
+        }
+        Ok(msgs)
+    }
+    // recv returns the Postgres message format and the Message. An error is
```

Comment:
Rustdoc convention, though way, way looser than godoc convention, would be "Returns the Postgres message format...". Specifically you don't include the name of the method as the first word in the description. It's also conventional that you always start with one short summary sentence and then a paragraph break, like you would in Python.

All together:

> Returns the PostgreSQL message format and the `Message`.
>
> If a new message is not received within the timeout, returns an error.

1000% optional; just noticed you were following godoc convention to a tee so figured I'd mention the equivalent rustdoc conventions.